pkg/util/http.go

```diff
@@ -129,7 +129,9 @@ func SerializeProtoResponse(w http.ResponseWriter, resp proto.Message, compressi
 	case NoCompression:
 	case FramedSnappy:
 		buf := bytes.Buffer{}
-		if _, err := snappy.NewWriter(&buf).Write(data); err != nil {
+		writer := snappy.NewBufferedWriter(&buf)
+		defer writer.Close()
```

Comment:
Don't we have to close the compressor before calling `buf.Bytes()` ?

(we try not to use FramedSnappy at all, but it's here for backwards-compatibility)

Reply:
That looks to be the case, this `Close()` call can not be deferred
```go
// The Writer returned buffers writes. Users must call Close to guarantee all
// data has been forwarded to the underlying io.Writer. They may also call
// Flush zero or more times before calling Close.
func NewBufferedWriter(w io.Writer) *Writer {
	return &Writer{
		w:    w,
		ibuf: make([]byte, 0, maxBlockSize),
		obuf: make([]byte, obufLen),
	}
}
```

Reply:
Fixed in 1c157c6